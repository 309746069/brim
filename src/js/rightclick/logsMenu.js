/* @flow */

import type {Space} from "../lib/Space"
import {
  countBy,
  detail,
  exclude,
  freshInclude,
  fromTime,
  groupByDrillDown,
  include,
  pcaps,
  toTime,
  whoisRightclick
} from "./actions"
import {hasGroupByProc} from "../lib/Program"
import Field, {TimeField} from "../models/Field"
import Log from "../models/Log"
import menuBuilder from "./menuBuilder"

export function logsMenu(program: string, space: Space) {
  return function(field: Field, log: Log) {
    const menu = menuBuilder()

    if (hasGroupByProc(program)) {
      menu.queryAction(groupByDrillDown(program, log))
    }

    if (field instanceof TimeField) {
      menu.queryAction(fromTime(field), toTime(field))
    } else {
      menu.queryAction(
        exclude(field),
        include(field),
        countBy(field),
        freshInclude(field)
      )
    }

    if (["addr", "set[addr]"].includes(field.type)) {
      menu.fieldAction(whoisRightclick(field))
    }

    if (log.isPath("conn") && space.packet_support) {
      menu.logAction(pcaps(log))
    }

    menu.logAction(detail(log))

    return menu.build()
  }
}
