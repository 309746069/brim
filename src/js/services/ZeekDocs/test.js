/* @flow */
import ZeekDocs from "./"

test("describe conn uid", () => {
  let desc = ZeekDocs.describe("conn", "uid")

  expect(desc).toBe("A unique identifier of the connection.")
})
