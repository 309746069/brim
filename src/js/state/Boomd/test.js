/* @flow */
import Boomd from "./"
import initTestStore from "../../test/initTestStore"

test("setting use analytics cache", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useCache(true))
  expect(Boomd.usingCache(store.getState())).toEqual(true)
  store.dispatch(Boomd.useCache(false))
  expect(Boomd.usingCache(store.getState())).toEqual(false)
})

test("setting use index", () => {
  const store = initTestStore()
  store.dispatch(Boomd.useIndex(true))
  expect(Boomd.usingIndex(store.getState())).toEqual(true)
  store.dispatch(Boomd.useIndex(false))
  expect(Boomd.usingIndex(store.getState())).toEqual(false)
})
