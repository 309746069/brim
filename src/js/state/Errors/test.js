/* @flow */

import AppError from "../../models/AppError"
import Errors from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("ERROR_CREATE", () => {
  let state = store.dispatchAll([Errors.createError("Bug")])

  expect(Errors.getErrors(state)[0]).toEqual(expect.any(AppError))
})

test("ERRORS_CLEAR", () => {
  let state = store.dispatchAll([
    Errors.createError("Bug"),
    Errors.createError("Bug2"),
    Errors.clearErrors()
  ])

  expect(Errors.getErrors(state)).toEqual([])
})
