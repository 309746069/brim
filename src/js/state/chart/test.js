/* @flow */

import chart from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let records = [
  [{name: "id", type: "string", value: "1"}],
  [{name: "id", type: "string", value: "2"}]
]

test("chart records append", () => {
  let state = store.dispatchAll([
    chart.appendRecords([records[0]]),
    chart.appendRecords([records[1]])
  ])

  expect(chart.getRecords(state)).toEqual(records)
})

test("chart records status", () => {
  let state = store.dispatchAll([chart.setStatus("SUCCESS")])

  expect(chart.getStatus(state)).toBe("SUCCESS")
})

test("chart records clear", () => {
  let state = store.dispatchAll([
    chart.appendRecords(records),
    chart.setStatus("SUCCESS"),
    chart.clear()
  ])

  expect(chart.getStatus(state)).toBe("INIT")
  expect(chart.getRecords(state)).toEqual([])
})
