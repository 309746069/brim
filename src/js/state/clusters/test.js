/* @flow */

import {addCluster, removeCluster, setCluster, setClusterError} from "./actions"
import {
  getCurrentCluster,
  getClusterMessage,
  getSavedClusters
} from "./selectors"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

let cluster = {
  host: "boom.com",
  port: "9867",
  username: "kerr",
  password: "123"
}

test("addCluster", () => {
  let state = store.dispatchAll([addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("addCluster when it already exists", () => {
  let state = store.dispatchAll([addCluster(cluster), addCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([cluster])
})

test("removeCluster", () => {
  let state = store.dispatchAll([addCluster(cluster), removeCluster(cluster)])

  expect(getSavedClusters(state)).toEqual([])
})

test("setCluster", () => {
  let state = store.dispatchAll([setCluster(cluster)])

  expect(getCurrentCluster(state)).toEqual(cluster)
})

test("setClusterMessage to message", () => {
  let state = store.dispatchAll([setClusterError("no spaces")])

  expect(getClusterMessage(state)).toEqual("no spaces")
})

test("setClusterMessage to empty", () => {
  let state = store.dispatchAll([setClusterError("")])

  expect(getClusterMessage(state)).toEqual("")
})
