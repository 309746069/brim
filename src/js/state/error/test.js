/* @flow */
import AppError from "../../models/AppError"
import error from "./"
import initTestStore from "../../test/initTestStore"

let store
beforeEach(() => {
  store = initTestStore()
})

test("init state", () => {
  let state = store.getState()
  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(false)
})

test("set", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e)])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(true)
})

test("clear", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.clear()])

  expect(error.getError(state)).toEqual(null)
  expect(error.getVisible(state)).toBe(true)
})

test("dismiss", () => {
  let e = new AppError()
  let state = store.dispatchAll([error.set(e), error.dismiss()])

  expect(error.getError(state)).toEqual(e)
  expect(error.getVisible(state)).toBe(false)
})
