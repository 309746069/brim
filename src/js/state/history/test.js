/* @flow */

import {changeSearchBarInput} from "../actions"
import History from "./"
import Tab from "../tab"
import initTestStore from "../../test/initTestStore"
import submitSearch from "../../flows/submitSearch"

let store
beforeEach(() => {
  store = initTestStore()
  store.dispatchAll([
    changeSearchBarInput("first"),
    submitSearch(),
    changeSearchBarInput("second"),
    submitSearch(),
    changeSearchBarInput("third"),
    submitSearch()
  ])
})

test("pushing history", () => {
  const state = store.getState()
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("third")
})

test("moving back changes the position", () => {
  let state = store.dispatchAll([History.back()])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("second")
})

test("going forward in history", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    History.forward()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("second")
})

test("going back in history then pushing new history", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    changeSearchBarInput("fourth"),
    submitSearch()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("fourth")
})

test("back, back, push, back", () => {
  let state = store.dispatchAll([
    History.back(),
    History.back(),
    changeSearchBarInput("fourth"),
    submitSearch(),
    History.back()
  ])
  const entry = Tab.currentEntry(state)
  expect(entry.program).toEqual("first")
})

test("clearing history", () => {
  let state = store.dispatchAll([History.clear()])
  expect(Tab.currentEntry(state)).toBe(undefined)
})
