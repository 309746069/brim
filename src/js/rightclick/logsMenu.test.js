/* @flow */

import {conn, dns, weird} from "../test/mockLogs"
import {logsMenu} from "./logsMenu"
import mockSpace from "../test/mockSpace"

describe("Log Right Click", () => {
  const space = mockSpace()
  const program = "*"

  test("conn log with pcap support", () => {
    const log = conn()
    const field = log.getField("id.orig_h")
    const menu = logsMenu(program, space)(field, log)

    expect(menu).toMatchSnapshot()
  })

  test("conn log without pcap support", () => {
    space.packet_support = false

    const log = conn()
    const field = log.getField("id.orig_h")
    const menu = logsMenu(program, space)(field, log)

    expect(menu).toMatchSnapshot()
  })

  test("dns log", () => {
    const log = dns()
    const field = log.getField("query")
    const menu = logsMenu(program, space)(field, log)

    expect(menu).toMatchSnapshot()
  })

  test("time field for weird log", () => {
    const log = weird()
    const field = log.getField("ts")
    const menu = logsMenu(program, space)(field, log)

    expect(menu).toMatchSnapshot()
  })

  test("time field for conn log", () => {
    const log = conn()
    const field = log.getField("ts")
    const menu = logsMenu(program, space)(field, log)

    expect(menu).toMatchSnapshot()
  })
})
