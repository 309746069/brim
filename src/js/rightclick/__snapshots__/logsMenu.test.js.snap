// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log Right Click conn log with pcap support 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Exclude this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Include this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Count by id.orig_h",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "New search with this value",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Whois Lookup",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Download PCAPS",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Open details",
    "type": "action",
  },
]
`;

exports[`Log Right Click conn log without pcap support 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Exclude this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Include this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Count by id.orig_h",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "New search with this value",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Whois Lookup",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Open details",
    "type": "action",
  },
]
`;

exports[`Log Right Click dns log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Exclude this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Include this value",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Count by query",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "New search with this value",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Open details",
    "type": "action",
  },
]
`;

exports[`Log Right Click time field for conn log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Use as \\"start\\" time",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Use as \\"end\\" time",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Open details",
    "type": "action",
  },
]
`;

exports[`Log Right Click time field for weird log 1`] = `
Array [
  Object {
    "onClick": [Function],
    "text": "Use as \\"start\\" time",
    "type": "action",
  },
  Object {
    "onClick": [Function],
    "text": "Use as \\"end\\" time",
    "type": "action",
  },
  Object {
    "type": "seperator",
  },
  Object {
    "onClick": [Function],
    "text": "Open details",
    "type": "action",
  },
]
`;
